### ISSUE TITLE
Version scheme bug in pre-release 0.7.1-beta1

### ISSUE THREAD

**Issue by benjaoming** [learningequality/kolibri#3127]
### Observed behavior

This issue is both concerned with a missing procedure instruction, execution, and an actual bug :)

1. Missing procedure execution:

After the release on 0.7.0, the branch was left at:

    VERSION = (0, 7, 0, 'final', 0)

This will mean that all releases think that they are final and skip the pre-release behavior for version detection -- both from VERSION files and git data.

2. Missing instruction 

After a release, the branch needs to be bumped to the next development stage. This should have been mentioned in the [Post-release TODO](http://kolibri-dev.readthedocs.io/en/develop/references/release_process.html#post-release-todo)

    VERSION = (0, 7, 1, 'alpha', 0)

3: Bug

Additionally, if you release with a `kolibri/VERSION` file with `0.7.1b1`, and it was based on the above `VERSION` tuple, we hit another problem:

```
Traceback (most recent call last):
  File ".virtualenvs/kolibri/bin/kolibri", line 9, in <module>
    load_entry_point('kolibri', 'console_scripts', 'kolibri')()
  File ".virtualenvs/kolibri/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 542, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File ".virtualenvs/kolibri/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2569, in load_entry_point
    return ep.load()
  File ".virtualenvs/kolibri/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2229, in load
    return self.resolve()
  File ".virtualenvs/kolibri/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2235, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/kolibri/__init__.py", line 10, in <module>
    __version__ = str(get_version(VERSION))
  File "/kolibri/utils/lru_cache.py", line 137, in wrapper
    result = user_function(*args, **kwds)
  File "/kolibri/utils/version.py", line 372, in get_version
    return get_prerelease_version(version)
  File "/kolibri/utils/version.py", line 343, in get_prerelease_version
    version_file
AssertionError: VERSION file inconsistent with kolibri.__version__. __version__ is: [0, 7, 1, u'alpha', 0], File says: 0.7.1b1
```

### Expected behavior

In our latest release, the version should be `0.7.1b1` because of the tag, but it reports as `0.7.0`.

Finally, this issue is also related to #2762 

### User-facing consequences

Makes life confusion for pre-release testers... so would suggest that we fix it for the beta2 just to ensure that stuff is working :)

### Errors and logs

n/a

### Steps to reproduce

n/a

### Context

n/a

**Comment by benjaoming** [comment]
Finally, this behavior hasn't been found in tests, because tox is running with `usedevelop=true` :/

**Comment by benjaoming** [comment]
1) The procedural issue is fixed #3133 
2) The documentation is updated #3116
3) The version logic is fixed #3136
